### DISCUSSION TITLE
Best way to get random rows by lucid

### DISCUSSION THREAD

**Discussion by DZ8540** [adonisjs/core#3874] [Help] [answered]
I wanted to find a similar discussions on this topic, but I could not find any of them. If there are, please post links to them. Thanks.

**Comment by Barbapapazes** [comment]
Hello, you can use the random function from SQL, order by "rand" and then select the first!

https://sql.sh/fonctions/rand

**Comment by DZ8540** [comment, marked as answer]
Thanks for answer from @Barbapapazes 
Link: https://github.com/adonisjs/core/discussions/3874#discussioncomment-3442802

BUT

In my case, worked 
```js
query.orderByRaw('RANDOM()')
```

RAND(), not worked.

RANDOM()
<img width="754" alt="image" src="https://user-images.githubusercontent.com/50152205/185940651-f9c28dcf-0913-42aa-8471-f8f3fd5f6c21.png">

RAND()
<img width="754" alt="image" src="https://user-images.githubusercontent.com/50152205/185940759-e069daff-d67c-4992-8192-54d6ec053f13.png">

**Reply by Barbapapazes** [reply]
Yes, in fact, PostgreSQL implements RANDOM but MySQL use RAND

PostgreSQL: https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-random-range/
MySQL: https://mysqlcode.com/mysql-rand/

**Comment by guoyunhe** [comment]
Since MySQL and Postgres has different implementation, I think it make sense to have a `random(seed?: number)` method for query builder, to make your code more portable.
